### PR TITLE
Ubuntu CI

### PR DESCRIPTION
**Pull request by markgoddard**

### CODE REVIEW

**Comment by markgoddard** [comment]
Got further - various tempest tests failed due to missing external network?

File: .github/workflows/stackhpc-pull-request.yml [hunk 1]
```diff
@@ -45,10 +45,10 @@ jobs:
       neutron_plugin: ovs
       OS_CLOUD: sms-lab-release
     secrets: inherit
-    if: github.repository == 'stackhpc/stackhpc-kayobe-config'
+    if: false && github.repository == 'stackhpc/stackhpc-kayobe-config'
```
Comment (markgoddard) [inline .github/workflows/stackhpc-pull-request.yml]: Need to test with the original condition

File: .github/workflows/stackhpc-pull-request.yml [hunk 2]
```diff
@@ -57,4 +57,32 @@ jobs:
...
+      - build-kayobe-image
+    uses: ./.github/workflows/stackhpc-all-in-one.yml
+    with:
+      kayobe_image: ${{ needs.build-kayobe-image.outputs.kayobe_image }}
+      os_distribution: ubuntu
+      neutron_plugin: ovn
+      vm_image: Ubuntu-20.04-new
+      OS_CLOUD: sms-lab-release
+    secrets: inherit
+    if: false && github.repository == 'stackhpc/stackhpc-kayobe-config'
```
Comment (markgoddard) [inline .github/workflows/stackhpc-pull-request.yml]: Need to remove `false &&`

File: .github/workflows/stackhpc-container-image-build.yml [hunk 3]
```diff
@@ -158,10 +158,15 @@ jobs:
...
+          filter='${{ inputs.regexes }}'
+          if [[ ${{ inputs.seed }} == 'true' ]]; then
+            filter="$filter bifrost"
+          fi
           gh workflow run \
           container-sync.yml \
           --repo stackhpc/stackhpc-release-train \
-          --ref main
+          --ref main \
+          -f filter="$filter"
```
Comment (markgoddard) [inline .github/workflows/stackhpc-container-image-build.yml]: This is kind of unrelated, perhaps it should be pulled out

File: .github/workflows/stackhpc-pull-request.yml [hunk 4]
```diff
@@ -57,4 +57,32 @@ jobs:
       neutron_plugin: ovn
       OS_CLOUD: sms-lab-release
     secrets: inherit
+    if: false && github.repository == 'stackhpc/stackhpc-kayobe-config'
```
Comment (markgoddard) [inline .github/workflows/stackhpc-pull-request.yml]: ditto

File: terraform/aio/scripts/configure-local-networking.sh [hunk 5]
```diff
@@ -9,50 +9,50 @@ cat << EOF | sudo tee -a /etc/hosts
 EOF
```
Comment (markgoddard) [inline terraform/aio/scripts/configure-local-networking.sh]: Apart from this part, this script is a noop. Will be interesting to see how CentOS copes with that.

File: terraform/aio/scripts/configure-local-networking.sh [hunk 6]
```diff
@@ -9,50 +9,50 @@ cat << EOF | sudo tee -a /etc/hosts
...
 # Configure an IP on the 'public' network to allow access to/from the cloud.
-if ! sudo ip a show dev breth1 | grep $public_ip/24 >/dev/null 2>&1; then
-  sudo ip a add $public_ip/24 dev breth1
-fi
+#if ! sudo ip a show dev breth1 | grep $public_ip/24 >/dev/null 2>&1; then
+#  sudo ip a add $public_ip/24 dev breth1
+#fi
 
 # This prevents network.service from restarting correctly.
 sudo killall dhclient || true
```
Comment (markgoddard) [inline terraform/aio/scripts/configure-local-networking.sh]: and this

File: terraform/aio/scripts/configure-local-networking.sh (hunk 5 above)
Comment (markgoddard) [reply terraform/aio/scripts/configure-local-networking.sh]: Actually we can drop this due to the userdata.cfg.tpl change

Comment (markgoddard) [reply terraform/aio/scripts/configure-local-networking.sh]: Oh, actually we've' dropped this file from the terraform config entirely.

**Review by markgoddard — COMMENTED** [review]
This patch adds Rocky Linux aio jobs: https://github.com/stackhpc/stackhpc-kayobe-config/pull/194/files. It was based on these changes, but evolved a bit. Let's merge that one first, then rework anything here to use a similar approach.

File: .github/workflows/stackhpc-pull-request.yml (hunk 1 above)
Comment (MoteHue) [reply .github/workflows/stackhpc-pull-request.yml]: Tests pass on all four aios

**Comment by markgoddard** [comment]
Rocky aio now merged

File: .github/workflows/stackhpc-container-image-build.yml (hunk 3 above)
Comment (markgoddard) [reply .github/workflows/stackhpc-container-image-build.yml]: Pulled out into https://github.com/stackhpc/stackhpc-kayobe-config/pull/239

**Comment by markgoddard** [comment]
I squashed the commits and pulled out the unrelated change. Should be good to go.

**Review by markgoddard — COMMENTED** [review]
LGTM

**Pull request merged**